src/main/java/com/metamx/metrics/JvmMonitor.java

```diff
@@ -176,23 +173,32 @@ void emit(ServiceEmitter emitter, Map<String, String[]> dimensions)
       }
     }
 
-    void emit(ServiceEmitter emitter, ServiceMetricEvent.Builder builder)
+    void emit(ServiceEmitter emitter, Map<String, String[]> dimensions)
     {
-      builder.setDimension("gcGen", name);
+      Map<String, String[]> dimensionsCopy = ImmutableMap
+          .<String, String[]>builder()
+          .putAll(dimensions)
+          .put("gcGen", new String[]{name})
+          .build();
 
       if (collector != null) {
-        collector.emit(emitter, builder);
+        collector.emit(emitter, dimensionsCopy);
```

Comment:
That's not correct because "gcName" dimension already is emitted for all spaces and is also emitted for collector itself. But if it's not so obvious I can separate it as you suggested into two different `if` statements